### PR TITLE
EvseManager: Wait for initial meter value

### PR DESCRIPTION
**Pull request by Pietfried**
## Describe your changes

If EVerest starts up with an EV plugged in (and authorization is present immediately), there is a race condition between the TransactionStarted event and the first powermeter value received. Without this change, transactions may start and use an initial meter value of 0.

This PR adds handling to wait for an initial powermeter value before the EvseManager becomes operational (is ready to start charging). The waiting time is configurable ( `initial_meter_value_timeout_ms` ) or can be disabled (by configuring 0)

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation
- [x] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements

### CODE REVIEW

**Comment by lategoodbye** [comment]
@Pietfried Is there a possible relation to this [issue](https://github.com/EVerest/libocpp/issues/678)?

File: modules/EvseManager/EvseManager.cpp [hunk 1]
```diff
@@ -677,6 +677,12 @@ void EvseManager::ready() {
                 latest_powermeter_data_billing = p;
             }
 
+            {
+                std::lock_guard<std::mutex> lg(powermeter_mutex);
```
Comment (corneliusclaussen) [inline modules/EvseManager/EvseManager.cpp:681]: I would prefer std::scoped_lock here as used in the rest of the module

File: modules/EvseManager/EvseManager.cpp [hunk 2]
```diff
@@ -893,6 +899,11 @@ void EvseManager::ready() {
         }
     });
 
+    // wait for first powermeter value
+    std::unique_lock<std::mutex> lk(powermeter_mutex);
```
Comment (corneliusclaussen) [inline modules/EvseManager/EvseManager.cpp:903]: Doesn't that deadlock if the first power meter value arrives after that lock? Maybe we should make initial_powermeter_value_received an std::atomic_bool instead and remove the locking in the power meter handler?

**Comment by Pietfried** [comment]
> @Pietfried Is there a possible relation to this [issue](https://github.com/EVerest/libocpp/issues/678)?

Good catch, looks like it. If the powermeter is not updated, the timestamp string defaults to `""`

File: modules/EvseManager/EvseManager.cpp (hunk 2 above)
Comment (Pietfried) [reply modules/EvseManager/EvseManager.cpp:903]: The `this->powermeter_cv.wait_for` releases the mutex so there should not be a deadlock

Comment (dorezyuk) [reply modules/EvseManager/EvseManager.cpp:903]: but you don't need the lock after the wait_for returns. Maybe its better to place both in an own scope

**Review by corneliusclaussen — APPROVED** [review]

**Pull request merged**